### PR TITLE
 Use the NIX_INDEX_DATABASE env var if possible.

### LINKED ISSUE #48: Permission denied error
Not sure if it's something related to [comma](https://github.com/nix-community/comma), but after #47 I had to remove the `comma` package from my `home-manager` config since I got

```
> error: collision between `/nix/store/l2x9kyb234qzrjyih36zfv2mk87m2b83-comma-with-db-1.6.0/bin/,' and `/nix/store/ch4s39pafpnm24fm9gpjqqpp494v8qm0-comma-1.6.0/bin/,'
```

Now although everything is fine, when I try to run any `comma` command I get

```
❯ , ls
error: creating directory '/nix/store/l2x9kyb234qzrjyih36zfv2mk87m2b83-comma-with-db-1.6.0/share/cache/nix': Permission denied
```

but using the "real" `comma` binary, it works as expected, i.e.:

```
/nix/store/ch4s39pafpnm24fm9gpjqqpp494v8qm0-comma-1.6.0/bin/comma ls
```

It looks like the overridden `XDG_CACHE_HOME` env variable in the wrapper is causing problems.  Am I missing something?

Thanks!

### LINKED ISSUE #58: use NIX_INDEX_DATABASE environment variable
if nix-index is at least [v0.1.6](https://github.com/bennofs/nix-index/releases/tag/v0.1.6),
we can make use of the environment variable introduced in https://github.com/bennofs/nix-index/pull/213

### PR DESCRIPTION
**Pull request by R-VdP**
Fixes #58.
Fixes #48.

### CODE REVIEW

File: nix-index-wrapper.nix [hunk 1]
```diff
@@ -8,10 +8,25 @@ runCommand "nix-index-with-db-${nix-index-unwrapped.version}"
...
   mkdir -p $out/share/cache/nix-index
   ln -s ${nix-index-database} $out/share/cache/nix-index/files
-  makeWrapper ${nix-index-unwrapped}/bin/nix-locate $out/bin/nix-locate \
-    --set XDG_CACHE_HOME $out/share/cache
+
+  version="$(${nix-index-unwrapped}/bin/nix-index --version | cut -d' ' -f2)"
+
+  if versionAtLeast "''${version}" "0.1.6"; then
+    makeWrapper ${nix-index-unwrapped}/bin/nix-locate $out/bin/nix-locate \
+      --set NIX_INDEX_DATABASE $out/share/cache
```
Comment (Mic92) [inline nix-index-wrapper.nix:25]: I think we need to point to `$out/share/cache/nix-index` here.

Comment (R-VdP) [reply nix-index-wrapper.nix:25]: From the source code (https://github.com/nix-community/nix-index/blob/2628e658385c3c8de53ae0c68bc9d7381d76b8e2/src/bin/nix-locate.rs#L242) I deduced that we point to the toplevel dir containing the `nix-index` dir.

It works in this way in any case, I tested it locally.

Comment (mmarx) [reply nix-index-wrapper.nix:25]: It does, indeed, not work:

```
❯ nix shell github:R-VdP/nix-index-database/nix_index_env_var
❯ nix-locate nix-locate
error: reading from the database at '/nix/store/7zpyhmrr1h5i77macc61rv2sc7p0408h-nix-index-with-db-0.1.6/share/cache/files' failed.
❯ NIX_INDEX_DATABASE=/nix/store/7zpyhmrr1h5i77macc61rv2sc7p0408h-nix-index-with-db-0.1.6/share/cache/nix-index /nix/store/gzwdnj0183s527d9gar4k4cvcjzmbg41-nix-index-0.1.6/bin/nix-locate nix-locate
nix-index.out                                         0 s /nix/store/0r2lyhsijmw6p0mqc7g2kk4rdjizwhvl-nix-index-0.1.5/bin/nix-locate
```

Comment (R-VdP) [reply nix-index-wrapper.nix:25]: Hmm, weird, I thought this worked on my machine. I'll check again tonight.

Comment (R-VdP) [reply nix-index-wrapper.nix]: Okay, fixed, and I did the same for comma too.

**Pull request merged**